### PR TITLE
[non_constant_identifier_names] check records

### LINKED ISSUE #3621: extend `non_constant_identifier_names` to include recordFields

### PR DESCRIPTION
**Pull request by pq**
Fixes #3621

(+Tests for #3622.)

/cc @scheglov @bwilkerson

### CODE REVIEW

**Comment by scheglov** [comment]
I'm not sure why you say that it fixes https://github.com/dart-lang/linter/issues/3622
Does it check also `RecordTypeAnnotations`?

**Comment by pq** [comment]
Sorry, no.  Thanks!

File: test/rules/non_constant_identifier_names_test.dart [hunk 1]
```diff
@@ -21,32 +21,32 @@ class NonConstantIdentifierNamesRecordsTest extends LintRuleTest {
   @override
   String get lintRule => 'non_constant_identifier_names';
 
-  @FailingTest(issue: 'https://github.com/dart-lang/linter/issues/3621')
   test_recordFields() async {
     await assertDiagnostics(r'''
 var a = (x: 1);
-var b = (X: 1);
+var b = (XXX: 1);
```
Comment (bwilkerson) [inline test/rules/non_constant_identifier_names_test.dart:27]: Do we need long names? (Just a different style than we follow in the other packages, where we keep the test cases as short as possible.)

**Review by bwilkerson — APPROVED** [review]

File: test/rules/non_constant_identifier_names_test.dart (hunk 1 above)
Comment (pq) [reply test/rules/non_constant_identifier_names_test.dart:27]: Just greater than 1.  I'll update.

**Pull request merged**